dkg/frost.go

```diff
@@ -0,0 +1,275 @@
+// Copyright © 2022 Obol Labs Inc.
+//
+// This program is free software: you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the Free
+// Software Foundation, either version 3 of the License, or (at your option)
+// any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of  MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along with
+// this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package dkg
+
+import (
+	"context"
+
+	"github.com/coinbase/kryptology/pkg/core/curves"
+	"github.com/coinbase/kryptology/pkg/dkg/frost"
+	"github.com/coinbase/kryptology/pkg/sharing"
+	"github.com/coinbase/kryptology/pkg/signatures/bls/bls_sig"
+
+	"github.com/obolnetwork/charon/app/errors"
+)
+
+// msgKey identifies the source and target nodes and validator index the message belongs to.
+type msgKey struct {
+	// ValIdx identifies the distributed validator (Ith parallel participant) the message belongs to.
+	ValIdx uint32
+
+	// SourceID identifies the source node/participant ID of the message.
+	// It is 1-indexed and equivalent to `cluster.NodeIdx.ShareIdx`.
+	SourceID uint32
+
+	// TargetID identifies the target node/participant ID of the message.
+	// It is 1-indexed and equivalent to `cluster.NodeIdx.ShareIdx`.
+	// The zero value indicates outgoing broadcast messages.
+	TargetID uint32
+}
+
+// fTransport abstracts the transport of frost DKG messages.
+type fTransport interface {
+	// Round1 returns results of all round 1 communication; the received round 1 broadcasts from all other nodes
+	// and the round 1 P2P sends to this node.
+	Round1(context.Context, map[msgKey]frost.Round1Bcast, map[msgKey]sharing.ShamirShare) (
```

Comment:
@corverroos is this an interface that another package is going to implement and inject? or all the implementations are in this package?

Either:
 + Round1, Round2 should be private
 + msgKey should be public

Reply:
It is a package internal interface. since the fTransport itself is not exported. Having unexported methods isn't required, nor does it makes things clearer.

Reply:
private is the default, if something are not going to be exported it should not be made public